### PR TITLE
Samples & Default of Enum

### PR DESCRIPTION
**Pull request by pksunkara**

### CODE REVIEW

File: test/schema.js [hunk 1]
```diff
@@ -713,6 +713,59 @@ describe('JSON Schema to Data Structure', () => {
...
+    expect(enumerations.get(1)).to.be.instanceof(StringElement);
+    expect(enumerations.getValue(1)).to.equal('two');
+
+    const samples = dataStructure.content.attributes.get('samples');
+
+    expect(samples.length).to.equal(2);
+    expect(samples.get(0).element).to.be.equal('enum');
+    expect(samples.get(0).toValue()).to.be.deep.equal('one');
+    expect(samples.get(1).element).to.be.equal('enum');
+    expect(samples.get(1).toValue()).to.be.deep.equal('two');
```
Comment (kylef) [inline test/schema.js]: I think it would be worth adding an additional check that `samples.get(1)`'s content is actually a string element vs a string directly. Since `toValue()` will return the string value regardless.

File: test/schema.js [hunk 2]
```diff
@@ -713,6 +713,59 @@ describe('JSON Schema to Data Structure', () => {
...
+    expect(enumerations.length).to.equal(2);
+    expect(enumerations.get(0)).to.be.instanceof(StringElement);
+    expect(enumerations.getValue(0)).to.equal('one');
+    expect(enumerations.get(1)).to.be.instanceof(StringElement);
+    expect(enumerations.getValue(1)).to.equal('two');
+
+    const defaultElement = dataStructure.content.attributes.get('default');
+
+    expect(defaultElement.element).to.be.equal('enum');
+    expect(defaultElement.toValue()).to.be.deep.equal('one');
```
Comment (kylef) [inline test/schema.js]: Ditto https://github.com/apiaryio/fury-adapter-swagger/pull/149/files#r153001597

File: src/schema.js [hunk 3]
```diff
@@ -202,16 +214,31 @@ export default class DataStructureGenerator {
         element.description = new StringElement(schema.description);
       }
 
-      if (schema.default !== undefined && !_.isArray(schema.default) &&
-          !_.isObject(schema.default)) {
+      let def = schema.default;
+
+      if (def !== undefined && !_.isArray(def) && !_.isObject(def)) {
         // TODO Support defaults for arrays and objects
```
Comment (kylef) [inline src/schema.js]: I think we should be able to remove this TODO and `isArray`/`isObject` check now, but we can do this in separate PR.

File: src/schema.js [hunk 4]
```diff
@@ -202,16 +214,31 @@ export default class DataStructureGenerator {
         element.description = new StringElement(schema.description);
       }
 
-      if (schema.default !== undefined && !_.isArray(schema.default) &&
-          !_.isObject(schema.default)) {
+      let def = schema.default;
```
Comment (kylef) [inline src/schema.js]: Can we call this `default` not `def` to remove the abbreviation? Clearer during reading. As a Pythonist `def` is shorthand for `define` too so doesn't feel right.

File: src/schema.js [hunk 5]
```diff
@@ -202,16 +214,31 @@ export default class DataStructureGenerator {
...
       }
 
+      let samples = [];
+
       if (schema.examples) {
-        element.attributes.set('samples', schema.examples);
+        samples = schema.examples;
       } else if (schema.example) {
-        element.attributes.set('samples', new ArrayElement([schema.example]));
+        samples = new ArrayElement([schema.example]);
```
Comment (kylef) [inline src/schema.js]: In other cases above the `samples` will be a JavaScript array instead of an `ArrayElement`. Can we make this consistent? The `ArrayElement` will get converted back to a normal array due to the `map` below too.

Little bit more confusing when `samples` can be either an array or `ArrayElement` and makes it easier to introduce bugs at a later date when it could be either types.

File: src/schema.js (hunk 4 above)
Comment (pksunkara) [reply src/schema.js]: `default` is reserved. Can't use it.
